src/counter.cpp

```diff
@@ -0,0 +1,59 @@
+#include "counter.h"
+#include <cstring>
+extern "C" {
+    #include <tree_sitter/api.h>
+}
+
+Counter::Counter(Parser* parser, Countable countable, std::string feature, std::vector<std::string>& files)
```

Comment:
You probably want to pass `countable` and `feature` by const-reference to minimize copying.  If you don't need to modify `files`, it should be passed by const reference as well.

[This](https://stackoverflow.com/a/5060218/6768191) Stack Overflow post does a good job explaining const reference stuff.